pom.xml

```diff
@@ -180,8 +180,8 @@
     <mysql.version>8.0.31</mysql.version>
     <postgres.version>42.7.3</postgres.version>
     <oracle.version>21.3.0.0</oracle.version>
-    <opencsv.version>2.3</opencsv.version>
-    <orc.version>1.8.5</orc.version>
+    <opencsv.version>5.9</opencsv.version>
```

Comment:
May I ask why do you change `opencsv` under the PR title, `HIVE-28224: Upgrade Orc version in Hive to 1.9.3`, @difin ?

Could you proceed to upgrade `opencsv` library independently before upgrading ORC?

Reply:
I added opencsv changes because Hive wasn't able to compile without them. Hive has a code that depends on opencsv version that was coming from Orc 1.8.5 and it is not in 1.9.3.

I created a separate PR for upgrading opencsv https://github.com/apache/hive/pull/5240
Once it gets merged, I will update this PR and remove opencsv changes from it.

Reply:
Thank you, @difin .

Reply:
You are welcome, @dongjoon-hyun

Reply:
Hi @dongjoon-hyun, opencsv changes were removed from this PR and were done separately. The differences in the results of tests are still present. 
I did a few additional checks: 
- The same differences are present when Hive statistics are enabled and when they are disabled. 
- With the `decimal_vgby` table used as a source in the test in Parquet format instead of Orc, the differences are absent (so it is an additional confirmation that it is something related to Orc).
- I dumped the content of the `decimal_vgby` table with and without Orc upgrade and they were the same, which is odd

Reply:
Hi @dongjoon-hyun,
I found that this Hive behavior changed as a result of [ORC-1266](https://issues.apache.org/jira/browse/ORC-1266).
It looks like this commit causes data correctness issues in Orc.
Can you please help?

Reply:
ORC-1266, at first, we call reset method every time we finish using `DecimalColumnVector`, to avoid inaccurate isRepeating and incorrect data reading, then we change to set `isRepeating=true` first, and then gradually traverse the data, if not repeated, then set `isRepeating=false`.
Later changed to first set to `isRepeating=true`, and then gradually traverse the data, if not repeated and then set `isRepeating=false`.

--- 

I did some validation testing on this vector_decimal_aggregate.q query and it has three values ​​that appear 1024 times, which is the same as the batch size read.
```
-119.4594594595
9318.4351351351
-4298.1513513514
```

Before ORC-1266, read 1024 identical values `-119.4594594595`, DecimalColumnVector's isRepeating is false, and vector has 1024 values.

After ORC-1266, because the value of the whole batch is the same, DecimalColumnVector's isRepeating is true, and vector has 1024 values.

This change should not be affected, because the read value has not changed.

--- 

Or I could submit a patch in ORC and not change it to true when isRepeating is set to false first.

Reply:
@cxzl25, how do we handle repeated nulls? it seems that `setIsRepeatingIfNeeded` is only called when `result.noNulls || !result.isNull[0]`?

Reply:
> Or I could submit a patch in ORC and not change it to true when isRepeating is set to false first.

See https://github.com/apache/orc/pull/1960

Reply:
@cxzl25 I did a test on my local env: backported your Orc patch into Orc branch 1.9 and run the Hive query test that had been failing previously. Now the test passed. Can you create a new Orc release from branch 1.9 so that we could use it in Hive? This is because we can't use Orc 2.0 or later because it requires Java 17+ and Hive still support Java 8.

Reply:
> how do we handle repeated nulls in ORC?

`isRepeating=true, noNulls=false, isNull[]=true`

<img width="471" alt="image" src="https://github.com/apache/hive/assets/3898450/b8ddde56-fccc-4661-920e-529a6f13272e">


> Also why logic is different from, for example, `DoubleColumnVector`, where we set `result.isRepeating = false;` at the beginning of `nextVector`

Because the formats for storing data are different, DecimalTreeReader and DoubleTreeReader behave differently.

org.apache.orc.impl.TreeReaderFactory.DecimalTreeReader#nextVector
```java
      // read the scales
      scaleReader.nextVector(result, scratchScaleVector, batchSize);
```